### PR TITLE
chore: add semantic-release versioning

### PR DESCRIPTION
**Pull request by ericcheatham**
# Description

This seeks to remove the need for manually creating releases (and thus triggering the action to publish this package). This PR uses [python-semantic-release](https://python-semantic-release.readthedocs.io/en/latest/) to automatically generate release PRs and cut github releases.

In order for this to work we will need to adhear to the Angular Style of commit messages. For our purposes that means we need to include tags in our PRs (e.g.: `chore`, `fix`, `feat`) 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature
- [ ] Bug fix
- [ ] Refactor
- [x] Documentation

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests
- [x] Manual Tests
- [ ] Deployed to staging

# Additional references

*Any additional links (if appropriate)*

### CODE REVIEW

**Comment by ericcheatham** [comment]
Need to make a decision on how/where we will be keeping the versioning information moving forward. Semantic release prefers a `__version__` variable somewhere in the package whereas we currently have `VERSION.txt` as our main source of truth

additionally we will need to add documentation around all of this because * gestures at all of this *

**Comment by ericcheatham** [comment]
Version is now maintained in the package init file. Semantic-release will be in charge of updating and changing this file.

File: MANIFEST.in [hunk 1]
```diff
@@ -4,5 +4,3 @@ include src/turbine/function-deploy/Dockerfile
 
 global-exclude *.py[co]
 global-exclude __pycache__
-
-include VERSION.txt
```
Comment (ericcheatham) [inline MANIFEST.in:8]: Removing the version file from the package manifest since it is no longer needed.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      - name: Checkout repository
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0
+      # packaging and dep management
+      - name: Install Python Semantic Release
+        run: python3 -m pip install python-semantic-release
+      - name: Prep release
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (ericcheatham) [inline .github/workflows/release.yml:27]: Semantic Release will create a commit and cut a release on the repo. In order to do that it needs a GH token

**Comment by ericcheatham** [comment]
I don't know what our appetite for having bots create commits and releases for us are.

File: setup.cfg [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [metadata]
 name = turbine-py
-version = file: VERSION.txt
+version = attr: turbine.__version__
```
Comment (janelletavares) [inline setup.cfg:3]: fancy

**Comment by janelletavares** [comment]
@ericcheatham should we also get rid of the publish* workflows in this PRs?

**Comment by ericcheatham** [comment]
@janelletavares I'm on the fence with that. 

The way I see it we have three options:
- Use python semantic versioning's publishing system. Under the hood it uses Tox to publish. 
- Continue to use our current publishing logic but fold it into one "stuff we do to release" yml. 
- leave it as is, the current publishing triggers when a release is created on the repo which semantic release does as a final step. 

As I think on it more I personally lean towards option 2. Primarily because we are also using the library that the Python Packaging Authority (they maintain PyPI) maintains for publishing. What are your thoughts?

**Comment by janelletavares** [comment]
> @janelletavares I'm on the fence with that.
> 
> The way I see it we have three options:
> 
> * Use python semantic versioning's publishing system. Under the hood it uses Tox to publish.
> * Continue to use our current publishing logic but fold it into one "stuff we do to release" yml.
> * leave it as is, the current publishing triggers when a release is created on the repo which semantic release does as a final step.
> 
> As I think on it more I personally lean towards option 2. Primarily because we are also using the library that the Python Packaging Authority (they maintain PyPI) maintains for publishing. What are your thoughts?

I can see leaving these around as a backup in case something fails with Tox. I would want that to be clearly documented, in the case that we're releasing while you're on vacation, etc.

**Comment by janelletavares** [comment]
Speaking of documentation... two of the three of these make commands do not exist as such https://github.com/meroxa/turbine-py#installation

**Review by janelletavares — APPROVED** [review]

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    steps:
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Checkout repository
+      - uses: actions/checkout@master
+        with:
+          token: ${{ secrets.SEMANTIC_RELEASE }}
```
Comment (ericcheatham) [inline .github/workflows/release.yml:21]: According to [the documentation](https://python-semantic-release.readthedocs.io/en/latest/automatic-releases/github-actions.html#example-workflow), Semantic Release is blocked from interacting with GH (specifically from creating releases) on protected branches. In order to get around that we are using a token form an admin on the repo (currently me, can be updated at a later time)

**Pull request merged**